python/pyspark/sql/session.py

```diff
@@ -161,8 +161,8 @@ def getOrCreate(self):
             with self._lock:
                 from pyspark.context import SparkContext
                 from pyspark.conf import SparkConf
-                session = SparkSession._instantiatedContext
-                if session is None:
+                session = SparkSession._instantiatedSession
+                if session is None or session._sc._jsc is None:
```

Comment:
We can add `isStopped` to `JavaSparkContext`, but I will prefer to solve it in Python side if possible.